src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -51,31 +69,67 @@ public void Dispose()
             // nop
         }
 
-        public Task StartAsync(CancellationToken cancellationToken)
+        public async Task StartAsync(CancellationToken cancellationToken)
         {
-            _poll = _folderSource.CreateFileWatcher(_fileWatcherType, OnFileWatcher, nextItem: null, pollIntervalInSeconds: _pollIntervalInSeconds > 0 ? _pollIntervalInSeconds : DefaultPollIntervalInSeconds);
+            CloudStorageAccount account;
+            if (!CloudStorageAccount.TryParse(this._config.StorageConnectionString, out account))
+            {
+                throw new ArgumentException(string.Format("The default storage account connection string is invalid: {0}", this._config.StorageConnectionString));
+            }
+
+            CloudBlobClient blobClient = account.CreateCloudBlobClient();
+
+            string apiHubBlobDirectoryPath = string.Format(ApiHubBlobDirectoryPathTemplate, this._siteName);
+            this._apiHubBlobDirectory = blobClient.GetContainerReference(HostContainerName).GetDirectoryReference(apiHubBlobDirectoryPath);
+
+            // Need to start polling from the point where it was left off. if this is the first time then lastPoll will be null.
+            var lastPoll = await GetLastPollStatusAsync();
+
+            Uri nextUri = null;        
+            Uri.TryCreate(lastPoll, UriKind.Absolute, out nextUri);
+
+            _poll = _folderSource.CreateFileWatcher(_fileWatcherType, OnFileWatcher, nextItem: nextUri, pollIntervalInSeconds: _pollIntervalInSeconds > 0 ? _pollIntervalInSeconds : DefaultPollIntervalInSeconds);
 
             if (_poll == null)
             {
-                var errorText = string.Format("Path '{0}' is invalid. IFolderItem.RootPath must be set to a valid directory location.", _folderSource.Path);
+                var errorText = string.Format("Path '{0}' is invalid. Path must be set to a valid folder location.", _folderSource.Path);
                 _trace.Error(errorText);
 
                 throw new InvalidOperationException(errorText);
             }
-
-            // TODO: need to decide what to do when _poll is null i.e. trigger folder does not exist.
-            return Task.FromResult(0);
         }
 
-        private Task OnFileWatcher(IFileItem file, object obj)
+        private async Task OnFileWatcher(IFileItem file, object obj)
         {
             ApiHubFile apiHubFile = new ApiHubFile(file);
-
+             
             TriggeredFunctionData input = new TriggeredFunctionData
             {
                 TriggerValue = apiHubFile
             };
-            return _executor.TryExecuteAsync(input, CancellationToken.None);
+
+            var functionResult = await _executor.TryExecuteAsync(input, CancellationToken.None);
+
+            if (functionResult.Succeeded)
+            {
+                string pollStatus = null;
+                var uri = obj as Uri;
```

Comment:
This obj comes from the callback being returned from CreateFileWatcher. if it is a real connector, it will be a Uri object and if it is against local files it will just be a DateTime object.